codersdk/workspaceagents.go

```diff
@@ -369,6 +373,7 @@ func (c *Client) DialWorkspaceAgent(ctx context.Context, agentID uuid.UUID, opti
 	firstDerpMap := make(chan error)
 	go func() {
 		defer close(closedDerpMap)
+		firstDerpMap := firstDerpMap // Shadowed so it can be reassigned outside goroutine.
```

Comment:
Why do we want to modify this from a separate goroutine? Is this not a potential data race?

Reply:
I put this here to safe-guard against data-race since we assign `firstDerpMap = nil` in the parent goroutine after we receive the first value (error or closed chan).

But I just realized, this is not needed. We only ever send one value (error or close, not both), so there is no race.

**Iff* we had not shadowed and done both: send error _and_ close the chan, then the `close(firstDerpMap)` would've been racy because we're reassigning in the other routine.